js/aes_gcm.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Subtle-based AES-GCM decryption supported on all browser types.
+ */
+
+/** 
+ * Possibly wrap an object in a Promise.
+ * @param {Object} op
+ * @return {!Promise}
+ */
+function wrapCryptoOp(op) {
+  if (typeof op.then == 'function') {
+    return op;
+  }
+  return new Promise(function (resolve, reject) {
+    op.oncomplete = function (e) {
+      resolve(op.result);
+    };
+    op.onerror = function (e) {
+      reject(e);
+    };
+  });
+}
+
+/**
+ * Interpret a byte array as a UTF-8 string.
+ * @param {!BufferSource} bytes
+ * @return {string}
+ */
+function utf8Decode(bytes) {
+  if (typeof TextDecoder !== 'undefined') {
+    return new TextDecoder('utf-8').decode(bytes);
+  }
+  const bytesBuffer = new Uint8Array(bytes.buffer || bytes);
+  const array = new Array(bytesBuffer.length);
+  for (let i = 0; i < bytesBuffer.length; i++) {
+    array[i] = String.fromCharCode(bytesBuffer[i]);
+  }
+  const asciiString = array.join('');
+  return decodeURIComponent(escape(asciiString));
+}
+
+/**
+ * Converts a base64 string into a Uint8Array with the corresponding bytes.
+ * @param {string} str
+ * @return {!Uint8Array}
+ */
+function base64Decode(str) {
+  const bytes = atob(str);
+  const len = bytes.length;
+  const array = new Uint8Array(len);
+  for (let i = 0; i < len; i++) {
+    const charCode = bytes.charCodeAt(i);
+    if (charCode >= 256) {
+      throw new Error("Decoded bytes not in range [0, 255].");
+    }
+    array[i] = charCode;
+  }
+  return array;
+}
+
+/**
+ * Decrypts the input text using AES-GCM with the input key.
+ * @param {string} key
+ * @param {string} text
+ * @return {!Promise}
+ */
+export function decryptAesGcm(key, text) {
+  const keybytes = base64Decode(key);
+  const isIE = !!window.msCrypto;
+  const subtle = (window.crypto || window.msCrypto).subtle;
```

Comment:
Let's replace `window` with `self` everywhere here. This way we get a worker-ready code.

Reply:
Done.